src/os/bluestore/BlueStore.cc

```diff
@@ -6555,7 +6567,15 @@ int BlueStore::_do_read(
               return r;
             return 0;
 	  });
-	assert(r == 0);
+        if (r < 0) {
+          derr << __func__ << " bdev-read failed: " << cpp_strerror(r)
+               << dendl;
+          if (r == -EIO) {
```

Comment:
assert(ioc->allow_eio) here?

Reply:
It will always be true since we explicitly set it at constructing (and is not allowed to change ever after) .
Actually this is the sync-read path, and that flag is only valid for truly aio-reads.